tests/xgboost/test_xgboost_autolog.py

```diff
@@ -238,6 +274,53 @@ def test_xgb_autolog_logs_metrics_with_early_stopping(bst_params, dtrain):
             assert metric_history == evals_result[eval_name][metric_name] + [best_metrics]
 
 
+@pytest.mark.large
+def test_xgb_autolog_batch_metrics_logger_logs_metrics_with_early_stopping(bst_params, dtrain):
+    patched_metrics_data = []
+
+    # Mock patching BatchMetricsLogger.record_metrics()
+    # to insure that expected metrics are being logged.
+    with patch(
+        "mlflow.utils.autologging_utils.BatchMetricsLogger.record_metrics"
+    ) as record_metrics_mock:
+
+        def record_metrics_side_effect(metrics, *args):
+            # pylint: disable=unused-argument
+            patched_metrics_data.extend(metrics.items())
+
+        record_metrics_mock.side_effect = record_metrics_side_effect
+
+        mlflow.xgboost.autolog()
+        evals_result = {}
+        params = {"eval_metric": ["merror", "mlogloss"]}
+        params.update(bst_params)
+        evals = [(dtrain, "train"), (dtrain, "valid")]
+        model = xgb.train(
+            params,
+            dtrain,
+            num_boost_round=20,
+            early_stopping_rounds=5,
+            evals=evals,
+            evals_result=evals_result,
+        )
+    metric_dict = dict(patched_metrics_data)
+    assert "best_iteration" in metric_dict
+    assert int(metric_dict["best_iteration"]) == model.best_iteration
+    assert "stopped_iteration" in metric_dict
+    assert int(metric_dict["stopped_iteration"]) == len(evals_result["train"]["merror"]) - 1
+
+    for eval_name in [e[1] for e in evals]:
+        for metric_name in params["eval_metric"]:
+            metric_key = "{}-{}".format(eval_name, metric_name)
+            metric_history = [x[1] for x in patched_metrics_data if x[0] == metric_key]
+
+            assert metric_key in metric_dict
+            assert len(metric_history) == 20 + 1
+
+            best_metrics = evals_result[eval_name][metric_name][model.best_iteration]
+            assert metric_history == evals_result[eval_name][metric_name] + [best_metrics]
```

Comment:
On second thought, I don't think we need to check the metric history this rigorously. Presumably, we do this in other test cases in the suite. Our goal here is just to make sure that `batch_metrics_logger` is being used; we can be reasonably confident that this is the case by comparing metric names in `patched_metrics_data`  entries with `run.data.metrics`.